flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/nodes/exec/serde/RexNodeJsonDeserializer.java

```diff
@@ -220,185 +260,298 @@ private Object toLiteralValue(
+                    }
+                    throw missingFunctionFromCatalog(identifier, false);
+                }
+            case ALL_ENFORCED:
+                {
+                    if (jsonNode.has(FIELD_NAME_CLASS)) {
+                        return deserializeFunctionClass(jsonNode, serdeContext);
+                    }
+                    final Optional<SqlOperator> lookupOperator =
+                            lookupOptionalSqlOperator(identifier, syntax, serdeContext, false);
+                    if (lookupOperator.map(RexNodeJsonDeserializer::isTemporary).orElse(false)) {
+                        return lookupOperator.get();
+                    }
+                    throw lookupDisabled(identifier);
+                }
+            case IDENTIFIER:
+                final Optional<SqlOperator> lookupOperator =
+                        lookupOptionalSqlOperator(identifier, syntax, serdeContext, true);
+                if (lookupOperator.isPresent()) {
+                    return lookupOperator.get();
+                } else {
+                    throw missingFunctionFromCatalog(identifier, true);
+                }
+            default:
+                throw new TableException("Unsupported restore strategy: " + restoreStrategy);
+        }
+    }
+
+    private static boolean isTemporary(SqlOperator sqlOperator) {
+        if (sqlOperator instanceof BridgingSqlFunction) {
+            return ((BridgingSqlFunction) sqlOperator).getResolvedFunction().isTemporary();
+        } else if (sqlOperator instanceof BridgingSqlAggFunction) {
+            return ((BridgingSqlAggFunction) sqlOperator).getResolvedFunction().isTemporary();
+        }
+        return false;
+    }
+
+    private static Optional<SqlOperator> lookupOptionalSqlOperator(
+            FunctionIdentifier identifier,
+            SqlSyntax syntax,
+            SerdeContext serdeContext,
+            boolean throwOnError) {
+        final List<SqlOperator> foundOperators = new ArrayList<>();
+        try {
+            serdeContext
+                    .getOperatorTable()
+                    .lookupOperatorOverloads(
+                            new SqlIdentifier(identifier.toList(), new SqlParserPos(0, 0)),
+                            null, // category
+                            syntax,
+                            foundOperators,
+                            SqlNameMatchers.liberal());
+            if (foundOperators.size() != 1) {
+                return Optional.empty();
+            }
+            return Optional.of(foundOperators.get(0));
+        } catch (Throwable t) {
+            if (throwOnError) {
+                throw new TableException(
+                        String.format("Error during lookup of function '%s'.", identifier), t);
             }
+            return Optional.empty();
+        }
+    }
+
+    private static TableException lookupDisabled(FunctionIdentifier identifier) {
```

Comment:
Also tests for those?

Reply:
tested in `TestRestoreIdentifier.TestRestoreAllEnforced.withConstantCatalogFunction`